src/modeler/AutoRest.Swagger/Validation/AvoidAnonymousTypes.cs

```diff
@@ -34,6 +34,18 @@ public class AvoidAnonymousTypes : TypedRule<SwaggerObject>
         public override string MessageTemplate => Resources.AnonymousTypesDiscouraged;
 
         /// <summary>
+        /// What kind of open api document type this rule should be applied to
+        /// </summary>
+        public override ServiceDefinitionDocumentType ServiceDefinitionDocumentType => ServiceDefinitionDocumentType.ARM;
```

Comment:
An openapi doc can have anonymous types in parameters, the restriction is ARM recommendation if I am not wrong